core/src/main/java/org/mule/runtime/core/api/artifact/ArtifactAstUtils.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.api.artifact;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableSet;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+import static org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType.OPERATION_DEF;
+import static org.mule.runtime.api.dsl.DslResolvingContext.getDefault;
+import static org.mule.runtime.ast.api.ArtifactType.APPLICATION;
+import static org.mule.runtime.core.api.util.boot.ExtensionLoaderUtils.getOptionalLoaderById;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_ARTIFACT_AST_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_EXTENSION_NAME_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_LOADER_ID;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_TYPE_LOADER_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.ExtensionConstants.VERSION_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.loader.ExtensionModelLoadingRequest.builder;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.api.ArtifactType;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.config.ConfigurationException;
+import org.mule.runtime.core.api.type.catalog.ApplicationTypeLoader;
+import org.mule.runtime.dsl.api.ConfigResource;
+import org.mule.runtime.extension.api.loader.ExtensionModelLoader;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+
+/**
+ * Utilities for parsing and handling {@link ArtifactAst}
+ *
+ * @since 4.5.0
+ */
+public final class ArtifactAstUtils {
```

Comment:
why would this be api? Who needs this outside of the runtime?

Reply:
because otherwise I get `NCD` when running tests